apps/site/assets/ts/models/alert.ts

```diff
@@ -159,24 +159,30 @@ const isCurrentLifecycle = ({ lifecycle }: Alert): boolean =>
   lifecycle === "ongoing" ||
   lifecycle === "ongoing_upcoming";
 
-export const isCurrentAlert = (
+export const isInNextXDays = (
   alert: Alert,
+  days: number,
   currentDate: Date = new Date()
 ): boolean => {
+  const xDays = add(Date.now(), { days });
```

Comment:
question: Should this be based on the `currentDate` argument instead of using `Date.now()`?

Reply:
They're interchangeable `currentDate` is Date object and `Date.now()` is the number representation

Reply:
But if we call `isInNextXDays(alert, 3, new Date(2023, 10, 10))` it'll return alerts that are within three days from whenever I call the function, as opposed to within three days from the date I specified.

Reply:
@thecristen  Thinking currentDate doesn't need to be a parameter but a variable instead. We only ever want alerts from now(the day the function is called) into sometime in the future

Reply:
@i0sea Rather true for our use case; but having it as a parameter can be quite powerful! it makes a few things possible:

- it makes the function easier to test: we can write tests for a specific `currentDate` value rather than its internal `Date.now()` changing based on whatever day the test runs.
- it lets us display the page according to a specific date.
  - we don't really exercise this in the frontend much... but one example where we support it is in the timetable backend - if you visit a `/timetable?date=2023-06-30`, the controller passes that `date` into the functions that ultimately generate the data needed, and it'll show the timetable for that date in lieu of defaulting to today.